File: superstitioMain/java/superstitio/customStrings/interFace/HasDifferentVersionStringSet.java
package superstitio.customStrings.interFace;

public interface HasDifferentVersionStringSet<T> {

    T getRightVersion();

    void initial();

    void initialSelfBlack();

    T getOriginVersion();

    void initialOrigin(T origin);

    Class<T> getSubClass();
}

File: superstitioMain/java/superstitio/customStrings/interFace/HasDifferentVersionStringSet.kt
package superstitio.customStrings.interFace

interface HasDifferentVersionStringSet<T>
{
    fun getRightVersion(): T

    fun initial()

    fun initialSelfBlack()

    val OriginVersion: T

    fun initialOrigin(origin: T)

    fun getSubClass(): Class<T>

    fun makeCopy(): HasDifferentVersionStringSet<T>
}

